docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -735,28 +563,86 @@ As database IDs are all globally unique, it's fine to replace table ID with data
+
+## Unresolved Questions
+
+### Compliance Requirements
+
+For compliance use-cases, it is clear that data at rest should reside within a geographic region. What is not clear is which (if any) circumstances data in transit is permitted to leave. There are several known issues which will need to be resolved:
 
-## Ecosystem tools
+* **Backup**: The `BACKUP` SQL command (and br) accept a single location such as `s3://bucket-location/my-backup` to centralize the backup. This centralization likely violates compliance requirements (assuming there are >=2 requirements that conflict on a single cluster). Backing up segments of data individually is both an inconsistent backup, and likely to result in misconfiguration which violates compliance rules. Users have a reasonable expectation of backup integration with compliance placement rules.
+* **CDC**: Similar to `BACKUP`, it should not be possible to subscribe to changes for data in a different jurisdiction.
+* **DDL**: The current implementation of DDL uses a centralized DDL owner which reads data from relevant tables and performs operations such as building indexes. The _read_ operation may violate compliance rules.
+* **Internal SQL**: Similar to DDL, several centralized background tasks, such as updating histograms/statistics need to be able to _read_ the data.
+* **DML**: It is not yet known which restrictions need to be placed on user queries. For example, if a poorly written user-query does not clearly target the `USA` partition when reading user data, should it generate an error because the `EUROPE` partition needs to be read in order for the semantics of the query to be correct? This may cause problems in development and integration environments if the restrictions can not be distilled into environments with smaller topologies.
 
-Many tools are based on binlog or metadata. For example, TiDB-binlog is based on binlog, while Lightning and Dumpling are based on metadata. Placement rules need to be compatible with these tools.
+### Behaviors
 
-If the downstream is not TiDB, no change needs to be made. But even if it is TiDB, TiKV nodes may have a different geographical topology, which means the labels of TiKV nodes may be different. In this case, placement rules can not be enforced on them.
+* The proposal says that `DROP PLACEMENT POLICY` should error if the policy is in use by any objects. This should be discussed, because it is not a common behavior in MySQL and may create some annoying debugging scenarios?
+* Are there risks to logical restore if creating a table with a non-existent `PLACEMENT POLICY` fails? This is the current proposal, but it can be changed to warning if it is an issue. If it can't be applied when creating the table, is the placement policy meta data dropped (and won't be picked up if that policy is created later?). This is regrettable, but seems like the most clear behavior because showing a non-defined `PLACEMENT POLICY` in `SHOW CREATE TABLE` is problematic.
```

Comment:
Yes to `Are there risks to logical restore`. Showing comments is fine, we have system tables and another show. Maybe provide a switch for `SHOW CREATE TABLE` to enable the output of placement rules, disable by default.

Reply:
We could do a switch for either dumping (modify SHOW CREATE TABLE output) or restoring (become less strict). MySQL has examples of both:

For dumping, there is syntax like "SHOW CREATE DATABASE IF NOT EXISTS db_name", which will actually add "IF NOT EXISTS" to the output. There are also some SQL modes which modify show create table, but it's usually unwise to add new modes.

For restoring, logical dumpers usually execute `SET FOREIGN_KEY_CHECKS=0` to skip constraint checking.

I'm not sure what's the better path. I'll have to think about it.